dvc/serialize.py

```diff
@@ -35,10 +36,10 @@ def _get_out(out):
     res = OrderedDict()
     if not out.use_cache:
         res[PARAM_CACHE] = False
-    if out.metric:
-        res[PARAM_METRIC] = True
-    elif out.persist:
```

Comment:
These are not really mutually exclusive, but our current schema for dvc.yaml doesn't support it. Plus our old dvcfiles were fine with it, so maybe worth just allowing it. Will revisit this line in this PR.